controllers/operator/apigateway_controller.go

```diff
@@ -246,7 +245,13 @@ func apiRulesExist(ctx context.Context, k8sClient client.Client) (bool, error) {
 }
 
 func oryRulesExist(ctx context.Context, k8sClient client.Client) (bool, error) {
-	oryRulesList := oryv1alpha1.RuleList{}
+	oryRulesList := unstructured.UnstructuredList{}
```

Comment:
Can we add a comment why we use the UnstructuredList?
```suggestion
         // To prevent reconciliation errors during Oathkeeper uninstallation, which arise from the absence of the Oathkeeper CRD,
	// we exclude calls to the rules from utilizing the cache. This can be achieved by using unstructured objects, as they are excluded from caching.
	oryRulesList := unstructured.UnstructuredList{}
```